ta/pkcs11/src/processing_asymm.c

```diff
@@ -903,3 +933,168 @@ enum pkcs11_rc do_asymm_derivation(struct pkcs11_session *session,
 
 	return rc;
 }
+
+static enum pkcs11_rc wrap_rsa_aes_key(struct active_processing *proc,
+				       void *data, uint32_t data_sz,
+				       void *out_buf, uint32_t *out_sz)
+{
+	enum pkcs11_rc rc = PKCS11_CKR_OK;
+	TEE_Result res = TEE_ERROR_GENERIC;
+	int mbedtls_rc = 0;
+	struct rsa_aes_key_wrap_processing_ctx *ctx = proc->extra_ctx;
+	mbedtls_nist_kw_context kw_ctx = { };
+	uint8_t aes_key_value[32] = { };
+	uint32_t aes_key_size = ctx->aes_key_bits / 8;
+	uint32_t aes_wrapped_size = *out_sz;
+	size_t target_key_size = 0;
+	const size_t kw_semiblock_len = 8;
+
+	if (ctx->aes_key_bits != 128 &&
+	    ctx->aes_key_bits != 192 &&
+	    ctx->aes_key_bits != 256)
+		return PKCS11_CKR_ARGUMENTS_BAD;
+
+	mbedtls_nist_kw_init(&kw_ctx);
+	TEE_GenerateRandom(aes_key_value, aes_key_size);
+	res = TEE_AsymmetricEncrypt(proc->tee_op_handle,
+				    NULL, 0,
+				    aes_key_value, aes_key_size,
+				    out_buf, &aes_wrapped_size);
+	if (res) {
+		if (res == TEE_ERROR_SHORT_BUFFER)
+			*out_sz = aes_wrapped_size + data_sz + kw_semiblock_len;
+
+		rc = tee2pkcs_error(res);
+		goto out;
+	}
+
+	if (*out_sz < aes_wrapped_size) {
+		rc = PKCS11_CKR_ARGUMENTS_BAD;
+		goto out;
+	}
+
+	mbedtls_rc = mbedtls_nist_kw_setkey(&kw_ctx, MBEDTLS_CIPHER_ID_AES,
+					    aes_key_value, ctx->aes_key_bits,
+					    true);
+	if (mbedtls_rc) {
+		rc = PKCS11_CKR_ARGUMENTS_BAD;
+		goto out;
+	}
+
+	mbedtls_rc = mbedtls_nist_kw_wrap(&kw_ctx, MBEDTLS_KW_MODE_KWP,
+					  data, data_sz,
+					  (uint8_t *)out_buf + aes_wrapped_size,
+					  &target_key_size,
+					  *out_sz - aes_wrapped_size);
+	if (mbedtls_rc) {
+		rc = PKCS11_CKR_ARGUMENTS_BAD;
+		goto out;
+	}
+
+	*out_sz = target_key_size + aes_wrapped_size;
```

Comment:
here we expect `target_key_size == data_sz + kw_semiblock_len`? otherwise the expect output size returned with `SHORT_BUFFER` is wrong. I suggest to add an assertion here:
`assert(target_key_size <= data_sz + kw_semiblock_len)`

Reply:
Here we expect
target_key_size <= data_sz + kw_semiblock_len

I suggest to add to check  if we want to make sure there is no overflow have occurred
*out_sz >= expected_size + target_key_size